File: exercism.io/javascript/high-scores/high-scores.js
//
// This is only a SKELETON file for the 'High-Scores' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class HighScores {
  constructor(scores) {
    this.points = scores
  }

  get scores() {
    return this.points
  }

  get latest() {
    return this.points[this.points.length - 1]
  }

  get personalBest() {
    return Math.max(...this.points)
  }

  get personalTopThree() {
    let p = [...this.points]
    return p.sort((a, b) => b - a).slice(0, 3)
  }
}

File: exercism.io/csharp/high-scores/HighScores.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class HighScores
{
    private List<int> list;
    private int length;
    public HighScores(List<int> list)
    {
        this.list = list;
        this.length = this.list.Count();
    }

    public List<int> Scores() => this.list;
    public int Latest() => this.list[this.length - 1];
    public int PersonalBest() => this.list.Max();
    public List<int> PersonalTopThree()
    {
        List<int> result = new List<int>(this.list);
        result.Sort();
        result.Reverse();
        int lenResult = this.length;
        while(lenResult > 3)
        {
            result.RemoveAt(lenResult - 1);
            lenResult--;
        }
        return result;       
    }
}